components/core/src/clp/ffi/utils.cpp

```diff
@@ -61,6 +61,12 @@ auto validate_and_append_escaped_utf8_string(std::string_view src, std::string&
                 constexpr uint8_t cLargestControlCharacter{0x1F};
                 auto const byte{static_cast<uint8_t>(*it)};
                 if (cLargestControlCharacter >= byte) {
+                    // NOTE: This section can be replaced with the following without losing any
+                    // performance:
+                    // escaped_char = fmt::format("\\u00{:02x}", byte);
+                    // We should update it once `fmtlib` is fully integrated into the FFI library.
+
+                    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
```

Comment:
```suggestion
                    // Necessary since implementations (besides `fmtlib`) that don't violate
                    // `cppcoreguidelines-pro-type-vararg` have much worse performance.
                    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
```

I think we should try to add comments above every nolint so that it's clear why we're turning it off.